src/chapters/style/documentation.typ

```diff
@@ -0,0 +1,247 @@
+
+Types in type lists or return type annotations may be separated by `|` to indicate that more than one type is accepted, the exact types allowed depend on the doc parser, but the built in types are generally supported.
+
+Parameter description and return types (if present) are placed tightly together, property annotations if present are separated using another empty doc comment line.
+
+Parameter documentation is created by writing a term item containing the parameter name, a mandatory type list in parentheses and optional description.
+If the parameter is an argument sink its name must likewise contain the spread operator `..`.
+Each parameter can only be documented once, but doesn't have to, undocumented parameters are considered private by convention and may only optional parameters.
+
+#do-dont[
+  ```typst
+  /// Function description
+  ///
+  /// / b (any): b descpription
+  /// / a (int | float): a description
+  #let func(a, b, c) = { ... }
+
+  /// Function description
+  ///
+  /// / ..args (int): args description
+  #let func(..args) = { ... }
+  ```
+][
+  ```typst
+  /// Function description
+  ///
+  /// / c (any): c doesn't exist
+  /// / a (int | float): a description
+  #let func(a, b) = { ... }
+
+  /// Missing empty line between function description and trailer
+  /// / b (any): b descpription
+  /// / a (int | float): a description
+  #let func(a, b) = { ... }
+
+
+  /// Function description
+  ///
+  /// / b (any): b descpription
+  /// / a: missing types
+  #let func(a, b) = { ... }
+
+  /// Function description
+  ///
+  /// / args (int): missing spread operator for args
+  #let func(..args) = { ... }
+  ```
+]
+
+The return type can only be annotated once, on a single line after all parameters, if any exist.
+For non function types the return type annotation can be used as a normal type annotation.
+Items which bind a `function.with` expression should be treated as regular function definitions, i.e. their return type is the return type of the function when called, not `function` itself as would be the return value of `function.with`.
+
+#do-dont[
+  ```typst
+  /// Function doc
+  ///
+  /// / arg (types): Description for arg
+  /// -> types
+  #let func(arg) = { ... }
+
+  /// Function doc
+  ///
+  /// -> types
+  #let other = func.with(default)
+
+  /// Value doc
+  ///
+  /// -> type
+  #let value = { ... }
+  ```
+][
+  ```typst
+  /// Missing empty line between description trailer
+  ///
+  /// -> type
+  /// / arg (types): arg descrption after return type
+  #let func(arg) = { ... }
+  ```
+]
+
+Property annotations can be used to document package specific or otherwise important information like deprecation status, visibility or contextuality.
+Such annotations may only be used after the return type (if one exists) and an empty doc comment line.
+
+#do-dont[
+  ```typst
+  /// Function doc
+  ///
+  /// / arg (types): Description for arg
+  /// -> types
+  ///
+  /// #property("deprecated")
```

Comment:
I'm wondering whether we can simply have predefined properties, so they could be simpler:

```typ
// #import tidy: *
#deprecated
#contextual
```

Since we have module, we can also use the properties with "namespace"s to disambiguate:

```typ
#tidy.deprecated
```

Reply:
I would argue this is an implementation detail, so I'd leave it up to the doc parsers and such to come up with a convention.

But we could define some defaults too, to avoid an ecosystem split from people choosing their own conventions.